cmd/conformance/main.go

```diff
@@ -0,0 +1,260 @@
+		case "--certificate-identity":
+			certSAN = &os.Args[i+1]
+			i += 2
+		case "--identity-token":
+			identityToken = &os.Args[i+1]
+			i += 2
+		case "--signature":
+			signaturePath = &os.Args[i+1]
+			i += 2
+		case "--trusted-root":
+			trustedRootPath = &os.Args[i+1]
+			i += 2
+		default:
+			i++
+		}
+	}
+}
+
+func main() {
+	if len(os.Args) < 2 {
+		usage()
+		os.Exit(1)
+	}
+
+	parseArgs()
+
+	args := []string{}
+
+	switch os.Args[1] {
+	case "sign":
+		args = append(args, "sign-blob")
+		if signaturePath != nil {
+			args = append(args, "--output-signature", *signaturePath)
+		}
+		if certPath != nil {
+			args = append(args, "--output-certificate", *certPath)
+		}
+		args = append(args, "-y")
+
+	case "sign-bundle":
+		args = append(args, "sign-blob")
+		args = append(args, "-y")
+
+	case "verify":
+		args = append(args, "verify-blob")
+
+		// TODO: for now, we handle `verify` by constructing a bundle
+		//
+		// Today cosign only supports `--trusted-root` with the new bundle
+		// format. When cosign supports `--trusted-root` with detached signed
+		// material, we can supply this content with `--certificate`
+		// and `--signature` instead.
+		fileBytes, err := os.ReadFile(os.Args[len(os.Args)-1])
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		fileDigest := sha256.Sum256(fileBytes)
+
+		pb := protobundle.Bundle{
+			MediaType: "application/vnd.dev.sigstore.bundle+json;version=0.1",
+		}
+
+		if signaturePath != nil {
+			sig, err := os.ReadFile(*signaturePath)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			sigBytes, err := base64.StdEncoding.DecodeString(string(sig))
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			pb.Content = &protobundle.Bundle_MessageSignature{
+				MessageSignature: &protocommon.MessageSignature{
+					MessageDigest: &protocommon.HashOutput{
+						Algorithm: protocommon.HashAlgorithm_SHA2_256,
+						Digest:    fileDigest[:],
+					},
+					Signature: sigBytes,
+				},
+			}
+		}
+		if certPath != nil {
+			cert, err := os.ReadFile(*certPath)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			pemCert, _ := pem.Decode(cert)
+			if pemCert == nil {
+				log.Fatalf("unable to load cerficate from %s", *certPath)
+			}
+
+			signingCert := protocommon.X509Certificate{
+				RawBytes: pemCert.Bytes,
+			}
+
+			pb.VerificationMaterial = &protobundle.VerificationMaterial{
+				Content: &protobundle.VerificationMaterial_X509CertificateChain{
+					X509CertificateChain: &protocommon.X509CertificateChain{
+						Certificates: []*protocommon.X509Certificate{&signingCert},
+					},
+				},
+			}
+		}
+
+		bundleFile, err := os.CreateTemp(os.TempDir(), "bundle.sigstore.json")
+		if err != nil {
+			log.Fatal(err)
+		}
+		bundleFileName := bundleFile.Name()
+		pbBytes, err := protojson.Marshal(&pb)
+		if err != nil {
+			log.Fatal(err)
+		}
+		if err := os.WriteFile(bundleFileName, pbBytes, 0600); err != nil {
+			log.Fatal(err)
+		}
+		bundlePath = &bundleFileName
+		args = append(args, "--insecure-ignore-tlog")
```

Comment:
Do conformance tests not include tests for Rekor?

Reply:
They do for `verify-bundle` but they do not for `verify`, because the only verification material the [conformance CLI protocol for `verify`](https://github.com/sigstore/sigstore-conformance/blob/main/docs/cli_protocol.md#verify) accepts is `--signature` and `--certificate`.